ui/src/app/routes.tsx

```diff
@@ -0,0 +1,265 @@
+import {ConceptSetDetails} from 'app/views/concept-set-details/component';
+import {ConceptSetList} from 'app/views/concept-set-list/component';
+import {DataPage} from 'app/views/data-page/component';
+import {DataSet} from 'app/views/dataset/component';
+import {Homepage} from 'app/views/homepage/component';
+import {NotebookList} from 'app/views/notebook-list/component';
+import {NotebookRedirect} from 'app/views/notebook-redirect/component';
+import {ProfilePage} from 'app/views/profile-page/component';
+import {Settings} from 'app/views/settings/component';
+import {SignIn} from 'app/views/sign-in/component';
+import {StigmatizationPage} from 'app/views/stigmatization-page/component';
+import {Unregistered} from 'app/views/unregistered/component';
+import {WorkspaceEdit, WorkspaceEditMode} from 'app/views/workspace-edit/component';
+import {WorkspaceList} from 'app/views/workspace-list/component';
+import {WorkspaceWrapper} from 'app/views/workspace-wrapper/component';
+import {Workspace} from 'app/views/workspace/component';
+
+const routes = [
+  {
+    name: 'login',
+    path: '/login',
+    render: () => <SignIn />,
+    title: 'Sign In',
+    public: true
+  },
+  {
+    name: 'homepage',
+    path: '/',
+    render: () => <Homepage />,
+    title: 'Homepage',
+  },
+  {
+    name: 'unregistered',
+    path: '/unregistered',
+    render: () => <Unregistered />,
+    title: 'Awaiting ID Verification'
+  },
+  {
+    name: 'stigmatization',
+    path: '/definitions/stigmatization',
+    render: () => <StigmatizationPage />,
+    title: 'Stigmatization Definition'
+  },
+  {
+    name: 'nihCallback',
+    path: '/nih-callback',
+    render: () => <Homepage />,
+    title: 'Homepage',
+  },
+  {
+    name: 'workspaces',
+    path: '/workspaces',
+    render: () => <WorkspaceList />,
+    title: 'View Workspaces',
+    breadcrumb: BreadcrumbType.Workspaces
+  },
+  {
+    name: 'workspace',
+    path: '/workspaces/:ns/:wsid',
+    render: () => <WorkspaceWrapper><Workspace /></WorkspaceWrapper>,
+    title: 'View Workspace Details',
+    breadcrumb: BreadcrumbType.Workspace
+  },
+  {
+    name: 'workspaceEdit',
+    path: '/workspaces/:ns/:wsid/edit',
+    render: () => <WorkspaceWrapper>
+      <WorkspaceEdit mode={WorkspaceEditMode.Edit}/>
+    </WorkspaceWrapper>,
+    title: 'Edit Workspace',
+    breadcrumb: BreadcrumbType.WorkspaceEdit
+  },
+  {
+    name: 'workspaceClone',
+    path: '/workspaces/:ns/:wsid/clone',
+    render: () => <WorkspaceWrapper>
+      <WorkspaceEdit mode={WorkspaceEditMode.Clone}/>
+    </WorkspaceWrapper>,
+    title: 'Clone Workspace',
+    breadcrumb: BreadcrumbType.WorkspaceClone
+  },
+  {
+    name: 'notebooks',
+    path: '/workspaces/:ns/:wsid/notebooks',
+    render: () => <WorkspaceWrapper><NotebookList /></WorkspaceWrapper>,
+    title: 'View Notebooks',
+    breadcrumb: BreadcrumbType.Workspace
+  },
+  {
+    name: 'notebook',
+    path: '/workspaces/:ns/:wsid/notebooks/:nbName',
```

Comment:
If the duplication of e.g. `/workspaces/:ns/:wsid` is a concern, it would be reasonable to extract that out into a variable, e.g. `workspacePrefix`.